src/rng.rs

```diff
@@ -37,32 +44,37 @@ impl<D: Digest> RngCore for FiatShamirRng<D> {
     }
 }
 
-impl<D: Digest> FiatShamirRng<D> {
+impl<D: Digest, R: RngCore + SeedableRng> FiatShamirRng for SimpleHashFiatShamirRng<D, R>
+where
+    R::Seed: From<[u8; 32]>,
+{
     /// Create a new `Self` by initializing with a fresh seed.
-    /// `self.seed = H(self.seed || new_seed)`.
+    /// `self.seed = H(initial_input)`.
     #[inline]
-    pub fn from_seed<'a, T: 'a + ToBytes>(seed: &'a T) -> Self {
+    fn initialize<'a, T: 'a + ToBytes>(initial_input: &'a T) -> Self {
         let mut bytes = Vec::new();
-        seed.write(&mut bytes).expect("failed to convert to bytes");
-        let seed = D::digest(&bytes);
-        let r_seed: [u8; 32] = FromBytes::read(seed.as_ref()).expect("failed to get [u32; 8]");
-        let r = ChaChaRng::from_seed(r_seed);
+        initial_input
+            .write(&mut bytes)
+            .expect("failed to convert to bytes");
+        let seed = FromBytes::read(D::digest(&bytes).as_ref()).expect("failed to get [u32; 8]");
```

Comment:
I think it is a previous typo: `"failed to get [u32; 8]")` to `"failed to get [u8; 32]")`, can you help fix it together (also exists on the next line) ?